clang/include/clang/CIR/Dialect/IR/CIRAttrs.td

```diff
@@ -772,21 +789,44 @@ def GlobalViewAttr : CIR_Attr<"GlobalView", "global_view", [TypedAttrInterface]>
 
   let parameters = (ins AttributeSelfTypeParameter<"">:$type,
                         "mlir::FlatSymbolRefAttr":$symbol,
-                        OptionalParameter<"mlir::ArrayAttr">:$indices);
+                        OptionalParameter<"mlir::ArrayAttr">:$indices,
+                        OptionalParameter<"ByteOffsetAttr">:$offset
```

Comment:
Can both `indices` and `offset` be used at the same time? Having two options is confusing, how would they relate to each other or how a user would pick up the right one to use?

Reply:
I see now that in the verifier you blocked this case, can you add some doc here to explain when is one supposed to be used versus the other? That might help me think more about the design.

Reply:
I'll also add that it might be better to (a) have only one optional attribute of the `Attribute` type, (b) make the verifier distinguish if they come from valid attributes with `isa<>`, (c) explain and give examples in the doc, and (d) add builder helpers.

Reply:
well, that's why I added a verifier for the `GlobalViewAttr` and check that they can't be used together.
The question about a user is a good one. I would say that  indices  should be used by default, and byte offset in the very rare cases like this one. But in the same time it implies that careful user should check both every time.   So may be it's not a good idea after all, what do you think?